src/Configuration.php

```diff
@@ -49,6 +49,22 @@ class Configuration
      **/
     private $xpath;
 
+     /**
+     * Boolean indicating if configuration results should be cached
+     *
+     * @access private
+     * @var    boolean
+     **/
+    private $useCaching;
+
+    /**
+     * The cache (used only when caching is enabled)
+     *
+     * @access private
+     * @var    array
+     **/
+    private $cache = ["alwaysArray" => [], "optionalArray" => [] ];
```

Comment:
I'm not going no argue about this. Every code style standard states consistency is the most important part and array() is already used in the library.

Apart from the fact that I doubt the array() construct is going to be deprecated anytime soon.